File: 2021/test/day_01_test.exs
defmodule Day01Test do
  use ExUnit.Case
  doctest Day01
  @moduletag :finished

  @sample_input """
  199
  200
  208
  210
  200
  207
  240
  269
  260
  263
  """

  @puzzle_input File.read!("priv/01.txt")

  test "part 1 example" do
    assert Day01.part_one(@sample_input) == 7
  end

  test "part 1 solution" do
    assert Day01.part_one(@puzzle_input) == 1482
  end

  test "part 2 example" do
    assert Day01.part_two(@sample_input) == 5
  end

  test "part 2 solution" do
    assert Day01.part_two(@puzzle_input) == 1518
  end
end

File: 2022/src/test/java/io/alephtwo/adventofcode/day/Day01Test.java
package io.alephtwo.adventofcode.day;

import io.alephtwo.adventofcode.util.InputLoader;
import java.nio.file.Paths;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day01Test {
    private static final Day DAY = new Day01();

    private static final String SAMPLE_INPUT =
            """
        1000
        2000
        3000

        4000

        5000
        6000

        7000
        8000
        9000

        10000
        """;

    private static final String PUZZLE_INPUT = InputLoader.loadInput(Paths.get("./input/01.txt"));

    @Test
    void partOneExamples() {
        assertEquals(24_000, DAY.partOne(SAMPLE_INPUT));
    }

    @Test
    void partOneSolution() {
        assertEquals(68_802, DAY.partOne(PUZZLE_INPUT));
    }

    @Test
    void partTwoExamples() {
        assertEquals(45_000, DAY.partTwo(SAMPLE_INPUT));
    }

    @Test
    void partTwoSolution() {
        assertEquals(205_370, DAY.partTwo(PUZZLE_INPUT));
    }
}
